docs/querying/sql-window-functions.md

```diff
@@ -0,0 +1,221 @@
+
+When working with window functions, consider the following:
+- Window functions only work on GROUP BY queries.
+- Window functions support aliasing.
+
+## Define a window with the OVER clause
+
+The OVER clause defines the query windows for window functions as follows:
+- PARTITION BY indicates the dimension that defines the rows within the window
+- ORDER BY specifies the order of the rows within the windows. Currently only ascending order, ASC, works.
+
+:::note
+
+Sometimes windows are called partitions. However, the partitioning for window functions are a shuffle (partition) of the result set that happens at query time and is not to be confused with Druid's segment partitioning feature which partitions data at ingest time.
+
+:::
+
+The following OVER clause example sets the window dimension to `channel` and orders the results by the absolute value of `delta` ascending:
+
+```sql
+...
+RANK() OVER (PARTITION BY channel ORDER BY ABS(delta) ASC)
+...
+```
+
+## Window function reference
+
+|Function|Notes|
+|--------|-----|
+| `ROW_NUMBER()`| Returns the number of the row within the window|
+|`RANK()`| Returns the rank for a row within a window | 
+|`DENSE_RANK()`| Returns the rank for a row within a window without gaps. For example, if two rows tie for rank of 1, the subsequent row is ranked 2. |
+|`PERCENT_RANK()`| Returns the rank of the row calculated as a percentage according to the formula: `(rank - 1) / (total window rows - 1)` |
+|`CUME_DIST()`| Returns the cumulative distribution of the current row within the window calculated as `number of window rows at the same rank or higher than current row` / `total window rows` |
+|`NTILE(tiles)`| Divides the rows within a window as evenly as possible into the number of tiles, also called buckets, and returns the value of the tile that the row falls into  |None |
+|`LAG(expr, offset, default)`| Returns the value evaluated at the row that precedes the current row by the offset number within the window; if there is no such row, returns the given default value |
+|`LEAD(expr, offset, default)`| Returns the value evaluated at the row that follows the current row by the offset number within the window; if there is no such row, returns the given default value |
+|`FIRST_VALUE(expr)`| Returns the value for the expression for the first row within the window|
+|`LAST_VALUE(expr)`| Returns the value for the expression for the last row within the window |
+
+## Examples
+
+The following example illustrates all of the built-in window functions to compare the number of characters changed per event for a channel in the Wikipedia data set.
+
+```sql
+SELECT FLOOR(__time TO DAY) AS event_time,
+    channel,
+    ABS(delta) AS change,
+    ROW_NUMBER() OVER w AS row_no,
+    RANK() OVER w AS rank_no,
+    DENSE_RANK() OVER w AS dense_rank_no,
+    PERCENT_RANK() OVER w AS pct_rank,
+    CUME_DIST() OVER w AS cumulative_dist,
+    NTILE(4) OVER w AS ntile_val,
+    LAG(ABS(delta), 1, 0) OVER w AS lag_val,
```

Comment:
I don't think this `default` work - I see `null` in the output; and not `0`
I assume this query can be run so I think we accept it ; but doesn't live up to working accordingly....

Reply:
@kgyrtkirk , don't know if this is a bug in the function or if there is not an intention to be a default value (I was basing my docs off pg)

Reply:
its a bug in the `LEAD/LAG` implementation ; it doesn't take the `default` into consideration; however `Calcite` accepts it - we could either reject these ; or implement it - either way I've opened [this ticket](https://github.com/apache/druid/issues/15270) for it